### PR TITLE
Fallback to counting lines when metrics fail ft 5698

### PR DESCRIPTION
**Pull request by bmbferreira**

### CODE REVIEW

**Comment by bmbferreira** [comment]
this PR depends on this: https://github.com/codacy/codacy-analysis-cli/pull/105

File: cli/src/main/scala/com/codacy/analysis/cli/command/analyse/MetricsToolExecutor.scala [hunk 1]
```diff
@@ -60,4 +65,43 @@ object MetricsToolExecutor {
...
+                nrMethods = None,
+                loc = countLoc(directory, file),
+                cloc = None,
+                complexity = None,
+                lineComplexities = Set.empty)
+            }
+        }(collection.breakOut)
+      }
+
+      fileMetrics.foreach(results => formatter.addAll(results.to[List]))
```
Comment (rtfpessoa) [inline cli/src/main/scala/com/codacy/analysis/cli/command/analyse/MetricsToolExecutor.scala]: calculateMissingFileMetrics doing the formatter does not seem a nice thing.

File: cli/src/main/scala/com/codacy/analysis/cli/command/analyse/MetricsToolExecutor.scala [hunk 2]
```diff
@@ -60,4 +65,43 @@ object MetricsToolExecutor {
...
       }(collection.breakOut)
   }
+
+  def calculateMissingFileMetrics(directory: File,
+                                  formatter: Formatter,
+                                  metricsResults: Seq[MetricsToolExecutorResult]): Seq[MetricsToolExecutorResult] = {
+
+    val analysisFiles: Set[Path] = metricsResults.flatMap(_.files)(collection.breakOut)
+
+    metricsResults.map { res =>
```
Comment (rtfpessoa) [inline cli/src/main/scala/com/codacy/analysis/cli/command/analyse/MetricsToolExecutor.scala]: Can we try something like
```
forEach file
  getMetrics
    case None => FileMetrics(loc = calculateLoc)
    case Some(metrics) if metrics.loc.isEmpty => FileMetrics(loc = calculateLoc)
    case Some(metrics) => metrics
```

File: cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala [hunk 3]
```diff
@@ -86,13 +86,23 @@ class AnalyseExecutor(toolInput: Option[String],
...
         result.partitionSubtypes[MetricsToolExecutorResult, IssuesToolExecutorResult, DuplicationToolExecutorResult]
-      MetricsToolExecutor.reduceMetricsToolResultsByFile(metricsResults) ++ issuesResults ++ duplicationResults
+      reduceAndCalculateMissingFileMetrics(metricsResults) ++ issuesResults ++ duplicationResults
     }
+
+    formatter.end()
+    analysisResultsWithMissingMetrics
+  }
+
+  private def reduceAndCalculateMissingFileMetrics(
```
Comment (rtfpessoa) [inline cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala]: Try to put this inside some post processing logic and then perform the reduce, missing and format inside.

**Comment by bmbferreira** [comment]
This PR depends on this: https://github.com/codacy/codacy-analysis-cli/pull/107

**Pull request merged**